### PR TITLE
fix lease interval value set

### LINKED ISSUE #183: If ManagedCluster.Spec.LeaseDurationSeconds is zero, will non-stop print event info.
If `ManagedCluster.Spec.LeaseDurationSeconds` not set. will non-stop print event info both with `hub` and `agent`, such as:
```
I1208 13:29:49.997717       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:50.796605       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:51.597894       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:52.396088       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:53.192194       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:53.994304       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:54.795829       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:55.592447       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:56.397188       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:57.196227       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:57.999585       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:58.794983       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:29:59.593935       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:00.402437       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:01.198800       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:02.000654       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:02.797852       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:03.602006       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:04.400387       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:05.224788       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:05.993830       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:06.791929       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:07.602048       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:08.394570       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:09.194666       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:09.999728       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:10.798960       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:11.595960       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:12.397709       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:13.195927       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:13.997736       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:14.811248       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:15.599120       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:16.397724       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:17.195343       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:17.999786       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:18.802863       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:19.596094       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:20.397046       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:21.196562       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:21.994035       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:22.791856       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:23.595581       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:24.400311       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:25.195641       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:25.996051       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:26.796031       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:27.596193       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:28.394320       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:29.198703       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:29.994557       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:30.797044       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:31.597022       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:32.396370       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:33.200775       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:34.002139       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:34.797702       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:35.598761       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:36.394833       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
I1208 13:30:37.200278       1 event.go:282] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"open-cluster-management", Name:"spoke-agent", UID:"39a61d5f-5f98-4c7b-b5c0-933bc4e25227", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'ManagedClusterStatusUpdated' the status of managed cluster "cluster1" has been updated, available condition is "True", due to "Managed cluster is available"
```

I think lease logic shouldn't depend on `admisionwebhook` set default value. Will set default value when `agent` create `ManagedCluster`.

### PR DESCRIPTION
**Pull request by champly**
Signed-off-by: champly <champly@outlook.com>

fix #183

### CODE REVIEW

**Comment by champly** [comment]
@mdelder @qiujian16 PTAL

File: pkg/hub/lease/controller.go [hunk 1]
```diff
@@ -95,6 +100,10 @@ func (c *leaseController) sync(ctx context.Context, syncCtx factory.SyncContext)
 			return err
 		case err == nil:
 			gracePeriod := time.Duration(leaseDurationTimes*cluster.Spec.LeaseDurationSeconds) * time.Second
+			// FIX: #183 avoid gracePeriod is zero, will non-stop update ManagedClusterLeaseUpdateStopped condition.
```
Comment (qiujian16) [inline pkg/hub/lease/controller.go:103]: could we comment here like:

```
set default grace period if it is zero in managedcluster
```
And I would also thinks this should be set as default in crd spec.

**Review by qiujian16 — COMMENTED** [review]
/approve

/assign @skeeey

File: pkg/hub/lease/controller.go (hunk 1 above)
Comment (champly) [reply pkg/hub/lease/controller.go:103]: this [pr](https://github.com/open-cluster-management-io/api/pull/111) fix this.

File: pkg/spoke/managedcluster/creating_controller.go [hunk 2]
```diff
@@ -68,6 +72,9 @@ func (c *managedClusterCreatingController) sync(ctx context.Context, syncCtx fac
 		ObjectMeta: metav1.ObjectMeta{
 			Name: c.clusterName,
 		},
+		Spec: clusterv1.ManagedClusterSpec{
+			LeaseDurationSeconds: int32(LeaseDurationSeconds),
```
Comment (skeeey) [inline pkg/spoke/managedcluster/creating_controller.go]: We have a [mutating webhook](https://github.com/open-cluster-management-io/registration/blob/main/pkg/webhook/cluster/mutating_webhook.go#L62) to set this value, and in your case, I guess you may not start the webhook, right? so I think we may not do this, what's your opinion?

Comment (champly) [reply pkg/spoke/managedcluster/creating_controller.go]: Yeal, i don't install webhook compoment. If set default values will not use this, i will remove this logic soon.

**Pull request merged**